src/main/java/com/r307/arbitrader/service/NotificationService.java

```diff
@@ -0,0 +1,118 @@
+package com.r307.arbitrader.service;
+
+import com.r307.arbitrader.config.NotificationConfiguration;
+import com.r307.arbitrader.service.model.Spread;
+import org.knowm.xchange.currency.Currency;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Inject;
+import java.math.BigDecimal;
+
+@Service
+@Async
+public class NotificationService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(NotificationService.class);
+    public static final String EMAIL_SUBJECT_NEW_ENTRY = "Arbitrader - New Entry Trade";
+    public static final String EMAIL_SUBJECT_NEW_EXIT = "Arbitrader - New Exit Trade";
+
+    private final JavaMailSender javaMailSender;
+    private final NotificationConfiguration notificationConfiguration;
+
+    @Inject
+    public NotificationService(JavaMailSender javaMailSender, NotificationConfiguration notificationConfiguration) {
+        this.javaMailSender = javaMailSender;
+        this.notificationConfiguration = notificationConfiguration;
+    }
+
+
+    public void sendEmailNotification(String subject, String body) {
+        if (!notificationConfiguration.getEmail().getActive()) {
```

Comment:
Make sure to check for nulls here too in case someone neglects to configure the feature or just comments it out in the configuration. I like the `active` flag for if you have a complete and correct configuration but you just want to turn it off (without deleting the config and losing that information) but I don't want the program to throw an NPE if the email configuration is missing - it's not critical for trading.